internal/pkg/comm/client.go

```diff
@@ -71,7 +71,9 @@ func (client *GRPCClient) parseSecureOptions(opts SecureOptions) error {
 
 	client.tlsConfig = &tls.Config{
 		VerifyPeerCertificate: opts.VerifyCertificate,
-		MinVersion:            tls.VersionTLS12} // TLS 1.2 only
+		MinVersion:            tls.VersionTLS12,
+		MaxVersion:            tls.VersionTLS13,
```

Comment:
Why explicitly set the max version instead of relying on the judgement of the tls package's calculation of max?

Reply:
I just preserved what was here, the fact the comment says "TLS 1.2 only" when tLS1.2 was the max, this meant setting the min version also forced the max inherently.

Should I remove the Max, it will keep us from having to perform upkeep on it in the future as new TLS versions become available, but it also means we will blindly accept the consequences of any new version of TLS?

I prefer the explicit declaration of a max supported TLS version, but I'm indifferent on what we decide.

Reply:
We should be able to remove the MaxVersion as gRPC  now works with TLS 1.3.
We can probably even remove the MinVersion as well given gRPC requires TLS 1.2 (it requires ALPN) but no harm explicitly setting the MinVersion just in case